benchmarking/src/llmperf/utils.py

```diff
@@ -153,11 +173,19 @@ def sample_random_positive_int(mean: int, stddev: int) -> int:
     """
 
     ret = -1
+    num_attempts = 0
     while ret <= 0:
         ret = int(random.gauss(mean, stddev))
+        num_attempts += 1
+        if num_attempts > NUM_RNG_ATTEMPTS:
+            warnings.warn(
+                f"Could not generate a random, positive integer after {NUM_RNG_ATTEMPTS} attempts. \n"
+                f"Check your choices for mean (currently: {mean}) and stddev (currently: {stddev}). "
+                f"Returning 1 instead."
+                )
+            return 1
```

Comment:
Could you please elaborate more why it should return 1 when number of attempts is reached?

Reply:
It's admittedly a total fake and kludge, designed to prevent an edge case where the original code could have run forever.
The original logic was to keep on drawing from the N(mu, sigma) distribution until a positive number is drawn. If mu was negative enough and sigma small enough, that process could go on forever: you'd NEVER draw the positive number you needed, and the `while` loop would never exit. That's silly, particularly since this is just a simple UI. So, I instead asked it to give up after 10 attempts or so and to then return the smallest permissible answer (1), with a warning.
It's probably academic here, but the same trick is used a bit later in the code when we build a prompt with lines of poetry. With a Llama-3 model, the base prompt length is 122 tokens long, and the UI allows the user to set mu as small as 50 and sigma as small as 10. 
The original logic would have kicked in to keep trying to draw from N(50,10) until a number >= 122 was drawn, which would never happen in the lifespan of the universe. So, the code again throws a warning and then just returns the shortest, base prompt with no poetry attached. 
I figured that was better than hanging in a forever loop with no idea why.

A lot of compromises arise because we're relying on an unbounded distribution (the Normal) in a context where there are in fact lower bounds on the numbers we can work with; happy to discuss other options.

Reply:
thanks for the details! It's clearer now 😄